### ISSUE TITLE
PluginTaskScheduler shouldn't store _next_ run timestamp, but _last_

### ISSUE THREAD

**Issue by grantila** [backstage/backstage#9755]
## Expected Behavior

When using the `PluginTaskScheduler` from `@backstage/backend-tasks` and scheduling tasks using `scheduleTask()`, I'd expect to be able to a) intentionally bump the `frequency` far away in the future, and b) make mistakes in the `frequency` setting and be able to revert that later. E.g. if I set it to 15 days instead of 15 minutes (maybe by mistake or intentionally to simply not have this run for a while, while testing/debugging/migrating/fixing) this won't work very well.

## Actual Behavior

Two problems occur:
 1. It won't respect the change until _next time_ since what's stored in the DB is the timestamp of the next run. So I can't set it "far away" in the future and ensure it won't run anytime soon.
 2. I won't be able to pull the frequency back from say 15 days to 15 minutes and have that happen more or less immediately. It can take 15 days(!). If I by mistake set it to 15 years and don't have manual access to the database in production; I'd have to deploy a hack that removes this row (which name/table should be a technical detail) and either deploy a revert quickly after or have some other fuzzy logic handle this somehow...

What I'd expect, is that the **last** timestamp of the run was stored in the database, and that {that + duration} will generate the next run. That'll mitigate these problems and be much more what a user would expect to happen.

**Comment by freben** [comment]
I'd prefer to keep the current structure, but writing a `MIN(old, new)` here instead https://github.com/backstage/backstage/blob/8cf64c4ec7552b20d64056aec15022590d991222/packages/backend-tasks/src/tasks/TaskWorker.ts#L142 .

**Comment by grantila** [comment]
> I'd prefer to keep the current structure

Why?

**Comment by freben** [comment]
Well OK let me rephrase that - I just want to preserve some properties. If they can be had with a refactor, then sure.

I wanted to be able to evolve and express the task timing info, in any way I see fit. You'll notice that it's versioned and verified. I want to be able to express complex schedules in ts code, but I also want to use the database server as the timing authority (the only clock source); I don't want js to have to load the tasks over and over again, parse them, do datetime arithmetic, and go back to sleep - if we can help it.

In `findReadyTask`, it uses SQL as `next_run_start_at <= ?` to find out if it's time to run yet, instead of loading and parsing the task description.

In `tryClaimTask`, it doesn't compute the next run timestamp, but rather the sql expression which computes it. Here's where I want to be able to have different future timing schedules and jitter and whatnot.

If we had a `last_run_start_at`, then sure we could perhaps also add a `dt` column that js computed and then in `findReadyTask` we could check something like `last_run_start_at + dt <= now()` or so ... But then maybe we'd want to leverage `current_run_started_at` instead and rename it? I dunno if worth.
